### ISSUE TITLE
Catch-all route doesn't work when static_folder ends in forwardslash

### ISSUE THREAD

**Issue by Sti2nd** [pallets/flask#3452]
I am trying to serve a React app with Flask. The React app uses React Router which means some routes should simply be handled by React. To achieve this one can usually use a catch-all route that returns the index.html. I first couldn't get the catch-all route to work at all, and then I got it working by removing the trailing slash for the `static_folder` parameter when calling Flask.

### Expected Behavior
When one enter a path with trailing forward slash in the `static_folder` parameter for flask.Flask() it shouldn't affect whether a catch-all route goes into effect or not.

In the below code remove the trailing slash for the `static_folder` and you are able to reach the catch-all route (home function) by entering a random route like `/something`.

```python
from flask import Flask, render_template

app = Flask(__name__, static_folder="frontend/build/", template_folder="frontend/build")

@app.route("/")
def route1():
    return "route1"


@app.route("/<path:path>")
def home(path):
    return "routeAny"
```

### Actual Behavior
Steps to reproduce
1. Run the above flask app
2. Open browser and verify that it renders the text "route1" for route `/`
3. Browse to `/something` and see that you get 404 Not Found.

### Environment

* Python version: 3.7.
* Flask version: 1.1.1
* Werkzeug version: 0.16.0

**Comment by Sti2nd** [comment]
I found out the reason that the static files didn't work without a trailing slash was because I serve index.html when requesting them. So I changed my code to serve the static files

```python
@app.route('/', defaults={'path': ''})
@app.route('/<path:path>')
def catch_all(path):
    build_folder = dir_path.joinpath("frontend", "build")
    if path != "" and os.path.exists(dir_path.joinpath(build_folder, path)):
        if path.count("/") > 1:
            [path, filename] = path.rsplit("/", maxsplit=1)
            return send_from_directory(dir_path.joinpath(build_folder, path), filename)
        else:
            filename = path
            return send_from_directory(dir_path.joinpath(build_folder), filename)
    else:
        return render_template("index.html")
```

Still don't understand why the static files worked with a trailing slash in the static_folder parameter. And I guess the bug still remains that a catch-all route won't catch anything when there is a trailing slash in the static_folder parameter.

**Comment by davidism** [comment]
Finally released 1.1.2 with this.
